include/envoy/config/config_provider.h

```diff
@@ -4,6 +4,7 @@
 
 #include "envoy/common/time.h"
 
+#include "common/common/assert.h"
 #include "common/protobuf/protobuf.h"
 
 #include "absl/types/optional.h"
```

Comment:
The term "provider" here is used within the context of the framework, where it denotes a ConfigProvider. I will clarify by using the actual type.

Reply:
Oh, did "xDS providers" refer to e.g. CdsApiImpl, LdsApiImpl, etc? It may be a matter of perspective, then. From where I've been working, those are consumers, from the parts of Envoy actually getting configured, they're providers.

Reply:
Yeah. The "provider" term must be viewed from the context of Envoy's non DS API logic that simply wants to consume the "implementation" of the config (i.e., the HCM simply wants to be able to get the RouteConfigProvider::ConfigConstSharedPtr that can then be used to get a route from (via Envoy::Router::Config::route())).